services/kaze/controllers/application.go

```diff
@@ -242,15 +245,23 @@ func FetchMetrics(c *gin.Context) {
 		}
 	}
 
-	metrics := mongo.FetchContainerMetrics(types.M{
-		mongo.NameKey: appName,
-		mongo.TimestampKey: types.M{
-			"$gte": time.Now().Unix() - timeSpan,
-		},
-	})
-
-	c.JSON(200, gin.H{
-		"success": true,
-		"data":    metrics,
+	chanStream := make(chan bool)
+	go func() {
+		defer close(chanStream)
+		chanStream <- true
+		time.Sleep(2 * configs.ServiceConfig.Mizu.MetricsInterval)
+	}()
+	c.Stream(func(w io.Writer) bool {
+		if _, ok := <-chanStream; ok {
```

Comment:
why is the `chanStream` part necessary?

You can just include the `time.Sleep` part in this function itself